AmplifyPlugins/DataStore/AWSDataStoreCategoryPluginTests/Sync/SubscriptionSync/ReconcileAndLocalSaveOperationTests.swift

```diff
@@ -141,19 +141,36 @@ class ReconcileAndLocalSaveOperationTests: XCTestCase {
         waitForExpectations(timeout: 1)
     }
 
-    func testExecuteApplyRemoteModel() throws {
+    func testExecuteApplyRemoteModelThatDoesNotExistLocally() throws {
         let expect = expectation(description: "action .execute applyRemoteModel")
         let disposition = RemoteSyncReconciler.Disposition.applyRemoteModel(anyPostMutationSync)
         storageAdapter.returnOnSave(dataStoreResult: .success(anyPostMutationSync.model))
         stateMachine.pushExpectActionCriteria { action in
-            XCTAssertEqual(action, ReconcileAndLocalSaveOperation.Action.applied(self.anyPostMutationSync))
+            XCTAssertEqual(action, ReconcileAndLocalSaveOperation.Action.applied(self.anyPostMutationSync,
+                                                                                 existsLocally: true))
             expect.fulfill()
         }
 
         stateMachine.state = .executing(disposition)
         waitForExpectations(timeout: 1)
     }
 
+    func testExecuteApplyRemoteModelThatExistsLocally() throws {
+        let expect = expectation(description: "action .execute applyRemoteModel")
+        let disposition = RemoteSyncReconciler.Disposition.applyRemoteModel(anyPostMutationSync)
+
+        storageAdapter.returnOnSave(dataStoreResult: .success(anyPostMutationSync.model))
+        storageAdapter.returnOnQueryMutationSyncMetadata(.some(anyPostMetadata))
+        stateMachine.pushExpectActionCriteria { action in
+            XCTAssertEqual(action, ReconcileAndLocalSaveOperation.Action.applied(self.anyPostMutationSync,
+                                                                                 existsLocally: false))
```

Comment:
`testExecuteApplyRemoteModelThatExistsLocally` is asserting `existsLocally: false`, see other comment about the production code should be 
`existsLocally = localMetadata != nil`

Reply:
updated